ops/charm.py

```diff
@@ -271,7 +375,7 @@ def add_metrics(self, metrics: typing.Mapping, labels: typing.Mapping = None) ->
             labels: {key:value} strings that can be applied to the
                 metrics that are being gathered
         """
-        self.framework.model._backend.add_metrics(metrics, labels)
+        self.framework.model._backend.add_metrics(metrics, labels)  # type:ignore
```

Comment:
strange that these ignores are necessary.  The types match :/

Reply:
those are probably for the private attr usage. Will replace with more targeted ignores